lib/config.js

```diff
@@ -499,7 +500,18 @@ AWS.Config = AWS.util.inherit({
    * @param [Constructor] dep A reference to a Promise constructor
    */
   setPromisesDependency: function setPromisesDependency(dep) {
-    AWS.util.addPromisesToRequests(AWS.Request, dep);
+    PromisesDependency = dep;
+    var constructors = [AWS.Request, AWS.Credentials, AWS.CredentialProviderChain];
+    if (AWS.S3 && AWS.S3.ManagedUpload) constructors.push(AWS.S3.ManagedUpload);
+    AWS.util.addPromises(AWS.Request, dep);
```

Comment:
Yes, thanks for catching that.
And for your question above about setting third-party promise dependencies when the environment supports native promises: all of the existing promisified methods will update to the set dependency because `setPromisesDependency` calls `AWS.util.addPromises` on the constructors (after I make the correction you mentioned above, of course). I've manually tested this. I can update one of the tests to be explicit about it.